libs/common/build.gradle

```diff
@@ -43,3 +43,51 @@ tasks.named('forbiddenApisMain').configure {
   // TODO: Need to decide how we want to handle for forbidden signatures with the changes to server
   replaceSignatureFiles 'jdk-signatures'
 }
+
+// Add support for incubator modules on supported Java versions.
+if (BuildParams.runtimeJavaVersion >= JavaVersion.VERSION_20) {
+  sourceSets {
+    java20 {
+      java {
+        srcDirs = ['src/main/java20']
+      }
+    }
+  }
+
+  configurations {
+    java20Implementation.extendsFrom(implementation)
+  }
+
+  dependencies {
+    java20Implementation sourceSets.main.output
+  }
+
+  compileJava20Java {
+    sourceCompatibility = JavaVersion.VERSION_20
+    targetCompatibility = JavaVersion.VERSION_20
+    options.compilerArgs += ['--add-modules', 'jdk.incubator.vector']
+    options.compilerArgs -= '-Werror' // use of incubator modules is reported as a warning
+  }
+
+  jar {
+    metaInf {
+      into 'versions/20'
+      from sourceSets.java20.output
+    }
+    manifest.attributes('Multi-Release': 'true')
+  }
+
+  test {
+    // Adds the multi-release JAR to the classpath when executing tests.
+    // This allows newer sources to be picked up at test runtime (if supported).
+    classpath += files(jar.archiveFile)
+    // Removes the "main" sources from the classpath to avoid JarHell problems as
+    // the multi-release JAR already contains those classes.
+    classpath -= sourceSets.main.output
+  }
+
+  forbiddenApisJava20 {
```

Comment:
It seems like `forbiddenApis` does not support incubator modules, I've looked (out of curiosity) how Apache Lucene folks handle that (tldr; they build extraction plugin) and weakening some checks sounds like the simplest path forward